Implement long running message

swatDong opened the pull request:
For https://github.com/microsoft/teams-ai/issues/219
Bot Builder .NET SDK supports passing `Activity` to `ContinueConversationAsync`, so no need to explicitly copy original activity.

singhk97 commented on dotnet/packages/Microsoft.Bot.Builder.M365/Microsoft.Bot.Builder.M365/Application.cs:
```diff
@@ -138,19 +143,27 @@ public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancel
 
                 // Remove @mentions
-                if (_options.RemoveRecipientMention && ActivityTypes.Message.Equals(turnContext.Activity.Type, StringComparison.OrdinalIgnoreCase))
+                if (this._options.RemoveRecipientMention && ActivityTypes.Message.Equals(turnContext.Activity.Type, StringComparison.OrdinalIgnoreCase))
```
nit: could you remove the `this` from `this._options` to remain consistent with the rest of the code

singhk97 replied:
Same for the other use cases

swatDong replied:
Removed. I also changed rule *IDE0009 (Add this or Me qualification)* to *IDE0003 (Remove this or Me qualification)* (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0003-ide0009) to keep consistent with existing code.

singhk97 commented on dotnet/packages/Microsoft.Bot.Builder.M365/Microsoft.Bot.Builder.M365/Application.cs:
```diff
@@ -159,19 +172,22 @@ public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancel
                 {
                     // Begin a new chain of AI calls
-                    await _ai.ChainAsync(turnContext, turnState); 
+                    _ = await this._ai.ChainAsync(turnContext, turnState);
```
Is there a reason we want to assign these calls to `_`

swatDong replied:
There's warning due to [*IDE0058 (Remove unnecessary expression value)*](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0058) . I've removed that rule to keep consistent with existing code - we do need expression calls in code.

swatDong replied:
Added IDE0058 back since it helps detect unnecessary calls.

singhk97 reviewed (commented):
Left a few nit pick comments. Looks great otherwise.

corinagum commented:
@swatDong could you look into the warnings that are existing on the files you made changes to? Once you've addressed that and Kavin's comments, feel free to merge your PR in. I see you've made changes since Kavin's review but I want to make sure you feel you're done before merging :)

corinagum commented on dotnet/packages/Microsoft.Bot.Builder.M365/Microsoft.Bot.Builder.M365.Tests/.editorconfig:
```diff
@@ -3,5 +3,4 @@
 dotnet_diagnostic.CA2007.severity = none # Do not directly await a Task
 dotnet_diagnostic.IDE0008.severity = none # Use explicit type instead of var
-dotnet_diagnostic.IDE0058.severity = none # Remove unnecessary expression value
 dotnet_diagnostic.VSTHRD111.severity = none # Use .ConfigureAwait(bool)
```
Are these like linting rules? Why are we removing this line?

swatDong replied:
Yep, added it back since it helps detect unnecessary calls.
BTW, still disable it for test project since in test there will be a lot `Assert...` calls.

swatDong commented:
> @swatDong could you look into the warnings that are existing on the files you made changes to? Once you've addressed that and Kavin's comments, feel free to merge your PR in. I see you've made changes since Kavin's review but I want to make sure you feel you're done before merging :)

@corinagum @singhk97 , I fixed existing warnings on the files I changed (*Application.cs*, *ActivityHandlerTests.cs*). There are a few line changed, please help review or let me know if any rule should be changed.

corinagum reviewed (approved):
Thanks so much for the updates @swatDong ❤️
